packages/mds-utils/utils.ts

```diff
@@ -595,6 +595,22 @@ export const START_NOW = now()
 export const START_TOMORROW = now() + (now() % days(1))
 export const START_ONE_MONTH_FROM_NOW = now() - (now() % days(1)) + days(30)
 
+/**
+ *
+ * @param arr1 First list to zip
+ * @param arr2 Second list to zip
+ * @returns A list of zipped & mapped values
+ *
+ * @example zip([1, 2, 3], ['a', 'b', 'c'], (a, b) => `${a}${b}`) => ['1a', '2b', '3c']
+ * Zips two lists of equal length, mapping each tuple to a new value.
+ */
+const zip = <T, U, R>(arr1: T[], arr2: U[], mapper: (x: T, y: U) => R) => {
```

Comment:
I wonder if there's a way I could do R inference here as opposed to allowing it in explicitly... Would be nice if there could be generics that don't allow the overrides via `foo<TypeA, TypeB, ...>()` and require the inputs to be strongly typed, but I don't think that's a thing in TS.